docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+isolation is going to be more complicated. The collocated deployment can be based on Kubernetes labels or namespaces, but 
+cross-clusters connections can be based on requester IP addresses, which is not always easy to discover. 
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
+The “from” element of ingress 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on “podSelectors” (labels based), and/or 
+“namespaceSelectors” (labels based) and/or “ipBlocks”. The last one permits incoming network traffic from predefined IP 
+address blocks.
+
+Current “FybrikApplicationSpec” has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of “clusterName” and “workloadSelector”. Unfortunately, based on this information we can 
+implement only one label based “podSelector”. Furthermore, for the backward compatibility of the current implementation 
+assumes that a Read scenario should have at least one label, which is not always true. 
+
+### Suggetions:
+
+- extend the “FybrikApplicationSpec.selector” element with an array of 
+[NetworkPolicyPeers](https://github.com/kubernetes/api/blob/59fcd23597fd090dba6b7e903eb0a8c9e8efb0a6/networking/v1/types.go#L183)
+this “NetworkPoliciesPeers” array will be an input element for the NP `from`element. This will restrict possible incoming
+connections to the relevant Fybrik module.
+- a blueprint will provide the ingress input data for chain connected modules.
+
+- deprecate the “workloadSelector” element, but continue to support it for backward compatibility. Process it as an 
+additional “podSelector” entry in the “NetworkPoliciesPeers” array
+
+- terminate the backward compatibility assumption that all Read flow should have at least a single workload label. 
+It was deprecated several Fybrik versions ago.
+
+- change the Flow field to be required instead of optional.
```

Comment:
I agree that this makes more sense, but this will have backward compatibility implications

Reply:
The Flow were implemented an year ago.

Reply:
Yes, the flows were implemented but if no flow is indicated in the spec then it is assumed to be "read".  If you make it required there are customers who will have to change their implementations

Reply:
My plan to validate it with our customers, this bullet is not crucial, bit NP.